### PR TITLE
Add unit tests for com.lrs.common.utils.encrypt.SHA

### PR DESCRIPTION
**Pull request by Braavos6**
I've analysed your codebase and noticed that `com.lrs.common.utils.encrypt.SHA` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

File: admin-common/pom.xml [hunk 1]
```diff
@@ -56,10 +64,15 @@
 			<artifactId>commons-io</artifactId>
 			<version>2.3</version>
 		</dependency>
+        <dependency>
```
Comment (petersear) [inline admin-common/pom.xml]: The added dependency does not line up with those already in pom.xml.

File: admin-common/pom.xml [hunk 2]
```diff
@@ -56,10 +64,15 @@
...
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
 
 
-
-	</dependencies>
```
Comment (petersear) [inline admin-common/pom.xml]: Looks like this was indented correctly already(?)

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.Base64;
+
+public class SHATest {
+
+    @Test
+    public void testEncryptSHA() throws Exception {
+        final byte[] data = {1, 2, 3};
+        String retval = Base64.getEncoder().encodeToString((SHA.encryptSHA(data)));
```
Comment (petersear) [inline admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: line > 80 chars

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.lrs.common.utils.encrypt;
+
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.Base64;
+
+public class SHATest {
+
+    @Test
+    public void testEncryptSHA() throws Exception {
```
Comment (petersear) [inline admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java:23]: Are the `throws` needed?

**Review by petersear — CHANGES_REQUESTED** [review]

File: admin-common/pom.xml (hunk 1 above)
Comment (Braavos6) [reply admin-common/pom.xml]: moved

File: admin-common/pom.xml (hunk 2 above)
Comment (Braavos6) [reply admin-common/pom.xml]: moved

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java (hunk 3 above)
Comment (Braavos6) [reply admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: altered

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java (hunk 4 above)
Comment (Braavos6) [reply admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java:23]: The method being tested throws and the test requires it. unhandled exception

File: admin-common/pom.xml [hunk 5]
```diff
@@ -57,6 +65,24 @@
 			<version>2.3</version>
 		</dependency>
 
+		<dependency>
```
Comment (petersear) [inline admin-common/pom.xml:68]: testng dependency?

File: admin-common/pom.xml [hunk 6]
```diff
@@ -57,6 +65,24 @@
...
+			<groupId>org.testng</groupId>
+			<artifactId>testng</artifactId>
+			<version>7.0.0-beta4</version>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
```
Comment (petersear) [inline admin-common/pom.xml]: JUnit dependency appears twice

**Review by petersear — CHANGES_REQUESTED** [review]
Looks better now that the white space issues have been resolved :-)

File: admin-common/pom.xml (hunk 5 above)
Comment (Braavos6) [reply admin-common/pom.xml:68]: Not sure how that got there :)

File: admin-common/pom.xml (hunk 6 above)
Comment (Braavos6) [reply admin-common/pom.xml]: removed

**Review by petersear — APPROVED** [review]

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Test
+    public void testEncryptSHAString() throws Exception {
+        Assert.assertEquals("", SHA.encryptSHA((String) null));
+        Assert.assertEquals("0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33",
+                SHA.encryptSHA("foo"));
+    }
+
+    @Test
+    public void testEncryptSHAStringString() throws Exception {
```
Comment (freddytuxworth) [inline admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: Are all of the `throws Exception`s needed? If so, could they throw something more specific?

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        Assert.assertEquals("", SHA.encryptSHA(null, null));
+        Assert.assertEquals("0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33",
+                SHA.encryptSHA("foo", "SHA"));
+        Assert.assertEquals("0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33",
+                SHA.encryptSHA("foo", "SHA-1"));
```
Comment (freddytuxworth) [inline admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: Add test with `SHA-256`

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.Base64;
+
+public class SHATest {
+
+    @Test
+    public void testEncryptSHA() throws Exception {
+        final byte[] data = {1, 2, 3};
+        String retval = Base64.getEncoder()
+                .encodeToString((SHA.encryptSHA(data)));
+        Assert.assertEquals("cDeAcZjCKn0rCAc3HXY3eahP388=", retval);
```
Comment (freddytuxworth) [inline admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: Instead of base64 encoding these bytes, you could just use `assertArrayEquals(new byte[]{...})`, there should be 20 bytes in the array.

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java (hunk 8 above)
Comment (freddytuxworth) [inline admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: Also add test that hashes something other than `foo`

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    @Test
+    public void testEncryptSHAString() throws Exception {
+        Assert.assertEquals("", SHA.encryptSHA((String) null));
+        Assert.assertEquals("0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33",
+                SHA.encryptSHA("foo"));
```
Comment (freddytuxworth) [inline admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: Add test for a non-empty string other than `foo`

**Review by freddytuxworth — CHANGES_REQUESTED** [review]

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java (hunk 8 above)
Comment (Braavos6) [reply admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: Done

Comment (Braavos6) [reply admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: Done

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java (hunk 7 above)
Comment (Braavos6) [reply admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: Done

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java (hunk 10 above)
Comment (Braavos6) [reply admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: Done.. used bar as well

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java (hunk 9 above)
Comment (Braavos6) [reply admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: Done

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+        Assert.assertEquals("0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33",
+                SHA.encryptSHA("foo", "SHA"));
+        Assert.assertEquals("0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33",
+                SHA.encryptSHA("foo", "SHA-1"));
+
+        String sha256foo =
+                "2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae";
+        Assert.assertEquals(sha256foo, SHA.encryptSHA("foo", "SHA-256"));
+
+        Assert.assertEquals("62cdb7020ff920e5aa642c3d4066950dd1f01f4d",
```
Comment (freddytuxworth) [inline admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: Might be cleaner to move all of the SHA results to class static variables at the top, like
```
private static final String SHA_256_foo = "2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae";
private static final String SHA_1_bar = "...
```
and then reference them in the code. Will reduce line overruns also.

**Review by freddytuxworth — APPROVED** [review]

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java (hunk 11 above)
Comment (Braavos6) [reply admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: Done

**Review by freddytuxworth — APPROVED** [review]

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+            "62cdb7020ff920e5aa642c3d4066950dd1f01f4d";
+    private static final String SHA_1_bar =
+            "62cdb7020ff920e5aa642c3d4066950dd1f01f4d";
+    private static final String SHA_256_bar =
+            "fcde2b2edba56bf408601fb721fe9b5c338d10ee429ea04fae5511b68fbf8fb9";
+
+    @Test
+    public void testEncryptSHA() throws Exception {
+        final byte[] data = {112, 55, -128, 113, -104, -62, 42, 125, 43, 8, 7
+                , 55, 29, 118, 55, 121, -88, 79, -33, -49};
```
Comment (petersear) [inline admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: The line-break should be after the comma rather than starting a line with ", "

**Review by petersear — CHANGES_REQUESTED** [review]

File: admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java (hunk 12 above)
Comment (Braavos6) [reply admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: .... I pushed this for an external PR today morning

Comment (freddytuxworth) [reply admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: You can still change it, the PR will change too

Comment (Braavos6) [reply admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: Done

Comment (petersear) [reply admin-common/src/test/java/com/lrs/common/utils/encrypt/SHATest.java]: I'd approved before this version of the changes was made. I caught that when going over one last time after other comments had been resolved.

**Review by petersear — APPROVED** [review]
